datalad/distribution/dataset.py

```diff
@@ -222,6 +222,8 @@ def get_dataset_handles(self, pattern=None, fulfilled=None, absolute=False,
 #        raise NotImplementedError("TODO")
 
     # TODO maybe needs to get its own interface
+    # TODO: issue: auto_add_changes isn't that easy. consider an annex repo with
+    #       with changes to a file in git. We would currently call 'annex add .'
```

Comment:
FTR: For an annex repo this is IMHO:

`git annex add .`

Deals with everything that is not already handled by Git directly. Followed by:

`git add .`

Does the rest.